vscode/webviews/chat/TextArea.tsx

```diff
@@ -101,6 +101,36 @@ export const TextArea: React.FunctionComponent<ChatUITextAreaProps> = ({
         [onKeyUp]
     )
 
+    const onPaste = useCallback(
+        (event: React.ClipboardEvent<HTMLTextAreaElement>): void => {
+            const { clipboardData } = event
+            const textarea = inputRef.current
+            if (clipboardData && textarea) {
+                const language = parseEditorData(clipboardData.getData('vscode-editor-data'))
+
+                if (language) {
+                    event.preventDefault()
+                    event.stopPropagation()
+
+                    const text = clipboardData.getData('text/plain')
+                    if (text.split('\n').length > 1) {
+                        const newText = '```\n' + text.replace(/\r\n/g, '\n') + '\n```\n'
+
+                        const startPos = textarea.selectionStart
+                        const endPos = textarea.selectionEnd
+                        textarea.value =
+                            textarea.value.substring(0, startPos) +
+                            newText +
+                            textarea.value.substring(endPos, textarea.value.length)
+                    }
```

Comment:
```suggestion
                    const text = clipboardData.getData('text/plain')
                    if (text.split('\n').length > 1) {
                        event.preventDefault()
                        event.stopPropagation()
                        const newText = '```\n' + text.replace(/\r\n/g, '\n') + '\n```\n'

                        const startPos = textarea.selectionStart
                        const endPos = textarea.selectionEnd
                        textarea.value =
                            textarea.value.substring(0, startPos) +
                            newText +
                            textarea.value.substring(endPos, textarea.value.length)
                    }
```

Otherwise single lines of code won't be pasted, I think?

Reply:
oh yeah sorry i added the second check later hah